datanode/datanode.go

```diff
@@ -432,40 +456,90 @@ func (d *dataNode) advertise() {
 
 	err = d.clusterServices.Advertise(ad)
 	if err != nil {
-		d.opts.InstrumentOptions().Logger().Fatalf("failed to advertise data node",
-			zap.String("id", d.hostID),
-			zap.Error(err))
+		d.logger.With("error", err.Error()).Fatalf("failed to advertise data node")
 	} else {
-		d.opts.InstrumentOptions().Logger().Info("start advertising datanode to cluster",
-			zap.String("id", d.hostID))
+		d.logger.Info("start advertising datanode to cluster")
 	}
 }
 
-// GetTableShard returns the table shard from the datanode
-func (d *dataNode) GetTableShard(table string, shardID uint32) (*memstore.TableShard, error) {
-	return d.memStore.GetTableShard(table, int(shardID))
+func (d *dataNode) addTable(table string) {
```

Comment:
any conflict or change with the WatchShardOwnershipEvents in recovery?

Reply:
in distributed mode, we do not call InitShards as we do in single instance mode under MemStore (in recovery.go)